neuralpredictors/layers/readouts/factorized.py

```diff
@@ -33,6 +56,12 @@ def __init__(
         self.outdims = outdims
         self.positive_weights = positive_weights
         self.constrain_pos = constrain_pos
+        self.positive_spatial = positive_spatial
+        if positive_spatial and constrain_pos:
+            logger.warning(
```

Comment:
Fabi or Edgar once told us in a code review sessions, that any warning that was triggered because the user did something that they should not, people usually would use `warnings.warn`; whereas warnings the user cannot do anything about would be triggered by `logger.warning`. So in this case it would mean to rather use `warnings.warn`